app/src/main/java/com/task/ui/component/login/LoginActivity.kt

```diff
@@ -2,27 +2,24 @@ package com.task.ui.component.login
 
 import android.content.Intent
 import android.os.Bundle
+import androidx.activity.viewModels
 import androidx.lifecycle.LiveData
 import com.google.android.material.snackbar.Snackbar
 import com.task.data.Resource
 import com.task.data.dto.login.LoginResponse
 import com.task.databinding.LoginActivityBinding
-import com.task.ui.ViewModelFactory
 import com.task.ui.base.BaseActivity
 import com.task.ui.component.recipes.RecipesListActivity
 import com.task.utils.*
-import javax.inject.Inject
+import dagger.hilt.android.AndroidEntryPoint
 
 /**
  * Created by AhmedEltaher
  */
-
+@AndroidEntryPoint
 class LoginActivity : BaseActivity() {
-    @Inject
-    lateinit var viewModelFactory: ViewModelFactory
 
-    @Inject
-    lateinit var loginViewModel: LoginViewModel
+    private val loginViewModel: LoginViewModel by viewModels()
```

Comment:
How did you manage to inject the viewModel with the default ViewModelFactory using Hilt given that It has "DataRepositorySource" as a dependency?! I can't see you using AbstractSavedStateViewModelFactory(owner, defaultArgs) as a factory for your viewModels

Reply:
`by viewModels() ` Kotlin property delegate,  does that for you for free